## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 471911`
Commit: [`86754ccfd667a6561052725ae76d2d5ffe3f2b32`](https://github.com/NixOS/nixpkgs/commit/86754ccfd667a6561052725ae76d2d5ffe3f2b32) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/86754ccfd667a6561052725ae76d2d5ffe3f2b32..pull/471911/head))
Merge: [`321e765278f1cc8bc5a1298b8c4b17cdca2f2093`](https://github.com/NixOS/nixpkgs/commit/321e765278f1cc8bc5a1298b8c4b17cdca2f2093)

Logs: https://github.com/magnetophon/nixpkgs-review-gha/actions/runs/20331650355


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>csound-qt</li>
  </ul>
</details>
<details>
  <summary>:x: 29 packages failed to build:</summary>
  <ul>
    <li>faust (faust2)</li>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>faust2sc</li>
    <li>faustPhysicalModeling</li>
    <li>faustlive</li>
    <li>guitarix</li>
    <li>kapitonov-plugins-pack</li>
    <li>liquidwar</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>open-music-kontrollers.mephisto</li>
    <li>tambura</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>csound</li>
    <li>cyclone</li>
    <li>foxdot (python313Packages.foxdot)</li>
    <li>foxdot.dist (python313Packages.foxdot.dist)</li>
    <li>gem</li>
    <li>helmholtz</li>
    <li>maxlib</li>
    <li>mrpeach</li>
    <li>pd-else</li>
    <li>puredata</li>
    <li>puremapping</li>
    <li>python314Packages.foxdot</li>
    <li>python314Packages.foxdot.dist</li>
    <li>solfege</li>
    <li>sonic-pi</li>
    <li>supercollider</li>
    <li>supercollider-with-plugins</li>
    <li>supercollider-with-sc3-plugins</li>
    <li>supercolliderPlugins.sc3-plugins</li>
    <li>supercollider_scel</li>
    <li>timbreid</li>
    <li>zexy</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>csound-qt</li>
  </ul>
</details>
<details>
  <summary>:x: 28 packages failed to build:</summary>
  <ul>
    <li>faust (faust2)</li>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>faust2sc</li>
    <li>faustPhysicalModeling</li>
    <li>faustlive</li>
    <li>guitarix</li>
    <li>kapitonov-plugins-pack</li>
    <li>liquidwar</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>mooSpace</li>
    <li>open-music-kontrollers.mephisto</li>
    <li>tambura</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>csound</li>
    <li>cyclone</li>
    <li>foxdot (python313Packages.foxdot)</li>
    <li>foxdot.dist (python313Packages.foxdot.dist)</li>
    <li>gem</li>
    <li>maxlib</li>
    <li>mrpeach</li>
    <li>pd-else</li>
    <li>puredata</li>
    <li>puremapping</li>
    <li>python314Packages.foxdot</li>
    <li>python314Packages.foxdot.dist</li>
    <li>solfege</li>
    <li>sonic-pi</li>
    <li>supercollider</li>
    <li>supercollider-with-plugins</li>
    <li>supercollider-with-sc3-plugins</li>
    <li>supercolliderPlugins.sc3-plugins</li>
    <li>supercollider_scel</li>
    <li>timbreid</li>
    <li>zexy</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>faust (faust2)</li>
    <li>faust2firefox</li>
    <li>faust2jackrust</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>puredata</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>faust2jackrust</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>puredata</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>faust (faust2)</li>
    <li>faust2firefox</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>faust2jackrust</summary>
<pre>Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/y4x5vgla1jzfjv5cvkx50j0j9gfm3xyi-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Updating Autotools / GNU config script to a newer upstream version: ./tools/faust2pd/config.guess
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: installPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installPhase&quot; }
Running phase: fixupPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;fixupPhase&quot; }
checking for references to /nix/build/nix-12603-3678016159/ in /nix/store/3685hfjrzh20x8xi7ndfkjjpm3wqh525-faust2jackrust-2.83.1...
patching script interpreter paths in /nix/store/3685hfjrzh20x8xi7ndfkjjpm3wqh525-faust2jackrust-2.83.1
/nix/store/3685hfjrzh20x8xi7ndfkjjpm3wqh525-faust2jackrust-2.83.1/bin/faust2jackrust: interpreter directive changed from &quot;#! /bin/bash -e&quot; to &quot;/nix/store/yi3x6xpilh0bq0bgb61imy6gpv8b1la4-bash-5.3p3/bin/bash -e&quot;</pre>
</details>
<details>
<summary>faust2ladspa</summary>
<pre>Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/y4x5vgla1jzfjv5cvkx50j0j9gfm3xyi-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Updating Autotools / GNU config script to a newer upstream version: ./tools/faust2pd/config.guess
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: installPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installPhase&quot; }
Running phase: fixupPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;fixupPhase&quot; }
checking for references to /nix/build/nix-12603-3678016160/ in /nix/store/rjppsj6pxhzcc617rszl90ig0q4jn3ja-faust2ladspa-2.83.1...</pre>
</details>
<details>
<summary>puredata</summary>
<pre>      |                  
d_soundfile.c:218:18: error: &#x27;strnlen&#x27; is only available on macOS 10.7 or newer [-Werror,-Wunguarded-availability]
  218 |     size_t len = strnlen(filename, size);
      |                  ^~~~~~~
/nix/store/1b9inc48934yzc0m2w8j9sbva9svqhfy-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:133:10: note: &#x27;strnlen&#x27; has been marked as being introduced in macOS 10.7 here, but the deployment target is macOS 10.6.0
  133 | size_t   strnlen(const char *__s1, size_t __n) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
      |          ^
d_soundfile.c:218:18: note: enclose &#x27;strnlen&#x27; in a __builtin_available check to silence this warning
  218 |     size_t len = strnlen(filename, size);
      |                  ^~~~~~~
  219 |     if (len + 4 &gt;= size)
  220 |         return 0;
  221 |     strcpy(filename + len, &quot;.txt&quot;);
      |                                    
2 errors generated.
make[2]: *** [Makefile:2394: pd-d_soundfile.o] Error 1
make[2]: Leaving directory &#x27;/nix/build/nix-12603-3678016155/pd-0.55-2/src&#x27;
make[1]: *** [Makefile:544: all-recursive] Error 1
make[1]: Leaving directory &#x27;/nix/build/nix-12603-3678016155/pd-0.55-2&#x27;
make: *** [Makefile:447: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/cs/csound/package.nix
+++ b/pkgs/by-name/cs/csound/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   libsndfile,
   libsamplerate,
@@ -35,6 +36,14 @@ stdenv.mkDerivation {
     rev = "2536da284dd70ec7272040cb0763f70ae57123c4";
     sha256 = "sha256-NDYltwmjBsX1DWCjy8/4cXMSl3/mK+HaQHSKUmRR9TI=";
   };
+
+  patches = [
+    # Fix typo that breaks build
+    (fetchpatch {
+      url = "https://github.com/csound/csound/commit/bb9bafcfa17a87d3733eda1e25a812fd0be08ac6.diff";
+      hash = "sha256-0M047uALPAoyQP0LbfBAybb2DWQ2/6QwZXxUjs1O1fE=";
+    })
+  ];
 
   cmakeFlags = [
     "-DBUILD_CSOUND_AC=0"

--- a/pkgs/development/interpreters/supercollider/default.nix
+++ b/pkgs/development/interpreters/supercollider/default.nix
@@ -49,6 +49,12 @@ mkDerivation rec {
       url = "https://github.com/supercollider/supercollider/commit/7d1f3fbe54e122889489a2f60bbc6cd6bb3bce28.patch";
       hash = "sha256-gyE0B2qTbj0ppbLlYTMa2ooY3FHzzIrdrpWYr81Hy1Y=";
     })
+
+    # Fixes the build with GCC 15
+    (fetchpatch {
+      url = "https://github.com/supercollider/supercollider/commit/edfac5e24959b12286938a9402326e521c2d2b63.patch";
+      hash = "sha256-8DNCO5VEX6V0Q29A/v5tFC7u835bwNHvcNlZzmS0ADg=";
+    })
   ];
 
   postPatch = ''

--- a/pkgs/development/interpreters/supercollider/plugins/sc3-plugins.nix
+++ b/pkgs/development/interpreters/supercollider/plugins/sc3-plugins.nix
@@ -23,6 +23,12 @@ stdenv.mkDerivation rec {
       url = "https://github.com/supercollider/sc3-plugins/commit/3dc56bf7fcc1f2261afc13f96da762b78bcbfa51.patch";
       hash = "sha256-lvXvGunfmjt6i+XPog14IKdnH1Qk8vefxplSDkXXXHU=";
     })
+
+    # Fix build with GCC 15
+    (fetchpatch2 {
+      url = "https://github.com/supercollider/sc3-plugins/commit/deaa55a7204bedf65a2000a463ae87a481bf3eb8.patch";
+      hash = "sha256-d8+4ZmedAwVt/AlU/YKqQF+80shEa8DiPnvMwJtW/RM=";
+    })
   ];
 
   strictDeps = true;

